### PR TITLE
Two more content registry interfaces

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
and made separate packages for interfaces that must be implemented by items or blocks

### CODE REVIEW

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+public interface PlayerBlockBreakListener {
+	/**
+	 * Determines if a player can break this block and collect its drops.
+	 * @param world The world that the block is being broken in
+	 * @param pos The block's position
+	 * @param player The player breaking the block.
+	 * @return Whether the block should drop anything when broken.
+	 */
+	boolean canHarvest(World world, BlockPos pos, PlayerEntity player);
```
Comment (hYdos) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:31]: block break event class should not have canharvest

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import net.minecraft.item.ItemStack;
+
+public interface UseTickable {
+	/**
+	 * Called each tick when the item is being used.
+	 * @param stack The {@link ItemStack} being used
+	 * @param player The {@link PlayerEntity} using the item
+	 * @param time The amount of time in ticks that the item is being used
+	 */
+	void useTick(ItemStack stack, PlayerEntity player, int time);
```
Comment (hYdos) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: is this needed?

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import net.fabricmc.fabric.api.content.registry.v1.item.UseTickable;
+
+@Mixin(PlayerEntity.class)
+public class MixinPlayerEntity {
+	@Shadow
+	public PlayerInventory inventory;
+
+	@Shadow
+	private ItemStack field_8356;
```
Comment (hYdos) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java:37]: would be nice to get yarn mappings for this

**Review by hYdos — CHANGES_REQUESTED** [review]
fix thank

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 1 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:31]: it doesn't

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:31]: this interface should be implemented by a `Block`

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: yes. my custom food spawn enderdragons when being eaten

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java (hunk 3 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java:37]: probably fixed in the big pr

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java:37]: it's `itemInUse`

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (hYdos) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: bit of a nieche usecase. find 2 more reasons to have it and ill resolve the conversation

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: my food makes the player immune to attackers when being eaten

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: my item spawns fireballs when being used

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 1 above)
Comment (hYdos) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:31]: PlayerBlockBreakListener is the class thats in. break != harvest. call it use if ur doing that

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java:29]: my item sends packets to the server when being used

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 1 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:31]: `PlayerBlockHarvestListener`?

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1.block;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public interface PlayerBlockBreakListener {
```
Comment (shedaniel) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java:23]: @FunctionalInterface

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (TheMadHau5) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java]: this is good usecase

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 4 above)
Comment (TheMadHau5) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java]: bruh

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java (hunk 3 above)
Comment (TheMadHau5) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java:37]: fix thank

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 1 above)
Comment (TheMadHau5) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java]: so... what's the resolution on this?

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java]: making the player immune and sending packets are not a good usecase?
but spawning fireballs and enderdragons are? *fr* too?

Comment (TheMadHau5) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java]: the opposite

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java (hunk 4 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/block/PlayerBlockBreakListener.java]: functional interface = bad

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java (hunk 3 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinPlayerEntity.java:37]: fixed in my yarrn pr

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java]: that's 2 and 4
not 1 and 3
1 and 3 are the same

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java]: yes
always fix my confusion

**Review by TheMadHau5 — APPROVED** [review]
passes my qc

**Review by lizelive — APPROVED** [review]

**Comment by hYdos** [comment]
shedaniels requested changes where met. merging

**Pull request merged**